pkg/operator/ceph/csi/csi.go

```diff
@@ -40,10 +41,22 @@ func ValidateAndConfigureDrivers(context *clusterd.Context, namespace, rookImage
 	}
 
 	if CSIEnabled() {
-		if err := startDrivers(context.Clientset, context.RookClientset, namespace, serverVersion, ownerInfo, v); err != nil {
-			logger.Errorf("failed to start Ceph csi drivers. %v", err)
-			return
+		csiLock.Lock()
+		for i := maxRetries - 1; i >= 0; i-- {
+			if err := startDrivers(context.Clientset, context.RookClientset, namespace, serverVersion, ownerInfo, v); err != nil && strings.Contains(err.Error(), "error reading ConfigMap") {
+				if i != 0 {
+					logger.Errorf("failed to start Ceph csi drivers, will retry starting csi drivers %d more times. %v", i, err)
```

Comment:
Let's just have a single error message, it will be rare to see the retries anyway. If it says "will retry 0 more times", it is actually accurate also. :)